app/models/job_completion_summary.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class JobCompletionSummary < ApplicationRecord
+  validates :extraction_id, presence: true, uniqueness: true
+  validates :extraction_name, presence: true
+  validates :error_type, presence: true
+  validates :error_details, presence: true
+  validates :error_count, presence: true, numericality: { greater_than_or_equal_to: 0 }
+
+  scope :by_error_type, ->(type) { where(error_type: type) }
+  scope :recent, -> { order(last_error_at: :desc) }
+  scope :recent_errors, -> { recent }
+  scope :with_errors, -> { where('error_count > 0') }
+
+  def self.group_by_error_type
+    group(:error_type).count
+  end
+
+  def self.total_errors
+    sum(:error_count)
+  end
+
+  def stop_condition?
+    error_type == 'stop condition'
```

Comment:
completion_type

also probably good to store the type as a symbol without a space, so like `:stop_condition`